cores/arduino/time.cpp

```diff
@@ -17,20 +19,25 @@ __attribute__((weak)) void yield() {
 }
 
 static FspTimer main_timer;
-
-static uint32_t _top_counter;
+// specifying these details as constants makes micros() faster !
```

Comment:
dear @aentiger,

Specifying the timer constants as compiler pre-defined macros is the most CPU efficient implementation.
Using variables will add extra CPU cycles because the value of the variable must be fetched from the memory.

I timed the speed of micros() for some alternative implementations:

The test consists of calling micros() twice and comparing the value.
This is done 10000 times.

The timing results are:
                                                                average    max
Uno R3                                                    3.6 us       24 us
Uno R4 1.0.2 faulty implementation       10.0 us       17 us (omitting the negative periods)
Uno R4 macros                                        3.7 us         7 us
Uno R4 variables integer calculations       4.0 us         8 us
Uno R4 variables floating point correct  12.6 us       20 us 

Considering that the R4 clock frequency is 3x faster than the R3, it is disappointing that the call to micros() in the current implementation is 3x slower. (ignoring the overflow bug). Just fixing the overflow bug  makes it even 20% more slower.

The macros implementation just matches the R3 performance. I think this is partly because the function call overhead of the R4 is more complex than on the R3: when defining the micros() function as an inline function, the performance is much better.

The difference between 3.7 us and 4.0 us seems small, but my application makes heavily use of the micros() function, so this 10% performance improvement of the macro implementation makes a significant difference for me.

 @facchinm raised a question on the support of the Portenta C33. I compiled and tested the test sketch on the Portenta and it turns out that a different used time.cpp is at different location: ~\hardware\renesas_portenta\1.0.2\cores\arduino. I will try to find an implementation that can be used for both boards.